client/src/ui/molecules/paginator/index.tsx

```diff
@@ -0,0 +1,48 @@
+import { Link } from "react-router-dom";
+import { range } from "../../../utils";
+
+function PageLink({ page }: { page: number }) {
+  return <Link to={`?page=${page}`}>{page}</Link>;
+}
+
+export function Paginator({
+  current,
+  last,
```

Comment:
Nit: Probably it would make sense to explicitly have `first = 0` here (and replace 0s by `first` in the implementation):

```suggestion
  first = 0,
  current,
  last,
```